catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -370,66 +420,121 @@ public NameIdentifier[] listSchemas(Namespace namespace) throws NoSuchCatalogExc
     }
   }
 
-  @Override
-  public Schema createSchema(NameIdentifier ident, String comment, Map<String, String> properties)
-      throws NoSuchCatalogException, SchemaAlreadyExistsException {
-    try {
-      if (store.exists(ident, Entity.EntityType.SCHEMA)) {
-        throw new SchemaAlreadyExistsException("Schema %s already exists", ident);
-      }
-    } catch (IOException ioe) {
-      throw new RuntimeException("Failed to check if schema " + ident + " exists", ioe);
-    }
+  /**
+   * Get the UserGroupInformation based on the NameIdentifier and properties.
+   *
+   * <p>Note: As UserGroupInformation is a static class, to avoid the thread safety issue, we need
+   * to use synchronized to ensure the thread safety: Make login and getLoginUser atomic.
+   */
+  private synchronized String initKerberos(
```

Comment:
I use the keyword `synchronized` to ensure that the login user and the user are put into the `userInfoMap` as the same user. 

Please help to check if it's okay;
@qqqttt123